#!/usr/bin/env python

##
## Copyright (C) 2017-2025 Hanson Robotics
##
## This program is free software: you can redistribute it and/or modify
## it under the terms of the GNU General Public License as published by
## the Free Software Foundation, either version 3 of the License, or
## (at your option) any later version.
##
## This program is distributed in the hope that it will be useful,
## but WITHOUT ANY WARRANTY; without even the implied warranty of
## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
## GNU General Public License for more details.
##
## You should have received a copy of the GNU General Public License
## along with this program.  If not, see <https://www.gnu.org/licenses/>.
##

PACKAGE = 'ros_chatbot'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("pull", bool_t, 0, "Pull Content", False)
gen.add("test", bool_t, 0, "Include Test Content?", False)
gen.add("automatic_pull", bool_t, 0, "Keep content up-to-date", False)
gen.add('pull_interval', int_t, 30, "The inverval in minutes for pulling content ", 30, 0, 1440)

# package name, node name, config name
exit(gen.generate(PACKAGE, "content_manager", "ContentManager"))
